src/v/archival/upload_housekeeping_service.cc

```diff
@@ -144,6 +149,8 @@ void upload_housekeeping_service::idle_timer_callback() {
         vlog(_ctxlog.debug, "Activating upload housekeeping");
         _workflow.resume(false);
     }
+
+    rearm_idle_timer();
```

Comment:
Also shouldn't be here I think. The housekeeping transitions into `active` state and idle timer gets armed. But the function of the idle timer is to transition the housekeeping into the `active` state. But since it's already active it doesn't make sense. It should be armed by the signal from the activity feed from the `remote`.

Reply:
Hmm. I think the call should be in an `else` block, but it's harmless.

The problem with relying on the notifications from remote is that no notifications come in when there's
no activity. That's precisely when we want to do stuff. This timer should re-arm itself independently for this reason.

Reply:
Looking at this again, it doesn't have to be in an else block. My mental model is as follows. `rearm_idle_timer` is called every `idle_timeout` in the background. When API activity crosses the threshold, the timer gets delayed by calling `rearm_idle_timer`.

Reply:
> But current implementation guarantees that the housekeeping will become active when there is no events from the remote after configured period of time.

I'm confused. By current implementation you mean not including this commit. I tried to rely on the housekeeping kicking in when cloud storage is idle and it didn't work previously (hence the change). Let's chat about it maybe.